cmd/kots/cli/deploy.go

```diff
@@ -0,0 +1,363 @@
+				return errors.Wrap(err, "failed to sync license")
+			}
+
+			// Step 2: If airgap bundle, push images first
+			if airgapBundle != "" {
+				if err := handleAirgapImagePush(v, clientset, appSlug, log); err != nil {
+					return errors.Wrap(err, "failed to push airgap images")
+				}
+			}
+
+			// Step 3: Upstream Update (both online and airgap)
+			if err := handleUpstreamUpdate(v, appSlug, localPort, authSlug, log); err != nil {
+				return errors.Wrap(err, "failed to process upstream update")
+			}
+
+			// Step 4: Set Config + Deploy
+			if err := handleSetConfigAndDeploy(v, appSlug, localPort, authSlug, log); err != nil {
+				return errors.Wrap(err, "failed to set config and deploy")
+			}
+
+			log.ActionWithoutSpinner("Deployment process completed successfully")
+
+			return nil
+		},
+	}
+
+	cmd.Flags().String("license", "", "path to license file (airgap mode only)")
+	cmd.Flags().String("channel-id", "", "channel ID")
+	cmd.Flags().Int64("channel-sequence", 0, "channel sequence")
+	cmd.Flags().String("airgap-bundle", "", "path to airgap bundle")
+	cmd.Flags().String("config-values", "", "path to config values file")
+	cmd.Flags().Bool("skip-preflights", false, "skip preflight checks")
+
+	cmd.MarkFlagRequired("config-values")
+
+	registryFlags(cmd.Flags())
+
+	return cmd
+}
+
+func handleLicenseSync(v *viper.Viper, appSlug string, localPort int, authSlug string, log *logger.CLILogger) error {
+	log.ActionWithoutSpinner("Syncing license...")
+
+	licenseData := ""
+
+	// Check if license flag was provided
+	if licenseFilePath := v.GetString("license"); licenseFilePath != "" {
+		data, err := os.ReadFile(licenseFilePath)
+		if err != nil {
+			return errors.Wrap(err, "failed to read license file")
+		}
+		licenseData = string(data)
+	}
+	// If no license provided, licenseData is empty and sync uses current license
+
+	requestPayload := map[string]interface{}{
+		"licenseData": licenseData,
+	}
+
+	requestBody, err := json.Marshal(requestPayload)
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal license sync request json")
+	}
+
+	url := fmt.Sprintf("http://localhost:%d/api/v1/app/%s/license", localPort, url.QueryEscape(appSlug))
+	newRequest, err := http.NewRequest("PUT", url, bytes.NewBuffer(requestBody))
+	if err != nil {
+		return errors.Wrap(err, "failed to create license sync http request")
+	}
+	newRequest.Header.Add("Authorization", authSlug)
+	newRequest.Header.Add("Content-Type", "application/json")
+
+	resp, err := http.DefaultClient.Do(newRequest)
+	if err != nil {
+		return errors.Wrap(err, "failed to execute license sync http request")
+	}
+	defer resp.Body.Close()
+
+	respBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return errors.Wrap(err, "failed to read license sync server response")
+	}
+
+	response := struct {
+		Success bool   `json:"success"`
+		Error   string `json:"error,omitempty"`
+		Synced  bool   `json:"synced"`
+	}{}
+	_ = json.Unmarshal(respBody, &response)
```

Comment:
can you handle this error always or at least for the successful case (if resp.StatusCode == http.StatusOK)